### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by troleksii**
- [DEMO LINK](https://troleksii.github.io/layout_product-cards/)
- [TEST REPORT LINK](https://troleksii.github.io/layout_product-cards/report/html_report/)

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -12,7 +12,42 @@
...
     />
   </head>
-  <body>
-    <h1>Product cards</h1>
+  <body class="body">
+    <div
+      class="card"
+      data-qa="card"
+    >
+      <div class="card__img">
```
Comment (Viktor-Kost) [inline src/index.html:20]: no needs to wrap `img` with `div`. Apply styles directly to `img` class selector

File: src/index.html [hunk 2]
```diff
@@ -12,7 +12,42 @@
...
+        <img
+          src="./images/imac.jpeg"
+          alt="iMac 27 Monoblock"
+          class="card__img-offer"
+        />
+      </div>
+      <div class="card__description">
+        APPLE A1419 iMac 27" Retina 5K Monoblock (MNED2UA/A)
+      </div>
+      <div class="card__code">Product code: 195434</div>
```
Comment (Viktor-Kost) [inline src/index.html:30]: use semantic tags for text content elements such as `h1-h6`, `p`, etc

File: src/index.html [hunk 3]
```diff
@@ -12,7 +12,42 @@
...
+      <div class="card__rate">
+        <div class="stars stars--4">
+          <div class="stars__star"></div>
+          <div class="stars__star"></div>
+          <div class="stars__star"></div>
+          <div class="stars__star"></div>
+          <div class="stars__star"></div>
+        </div>
+        <div class="card__reviews">Reviews: 5</div>
```
Comment (Viktor-Kost) [inline src/index.html:39]: semantic

File: src/styles/card.scss [hunk 4]
```diff
@@ -0,0 +1,96 @@
+@font-face {
+  font-family: Roboto;
+  src: url('https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,100;0,300;0,400;0,500;0,700;0,900;1,100;1,300;1,400;1,500;1,700;1,900&display=swap');
+}
+
+.body {
+  margin: 20px;
+}
```
Comment (Viktor-Kost) [inline src/styles/card.scss:8]: no needs
```suggestion
```

File: src/styles/card.scss [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+.card {
+  border: 1px solid #f3f3f3;
+  width: 200px;
+  border-radius: 5px;
+  box-sizing: border-box;
```
Comment (Viktor-Kost) [inline src/styles/card.scss:14]: it's allowed to use only this rule at `*` selector

File: src/styles/card.scss [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+  background: #fff;
+  padding: 32px 16px 16px;
+  font-family: Roboto, sans-serif;
```
Comment (Viktor-Kost) [inline src/styles/card.scss:17]: move this rule to `html` tag selector

NOTE!
Move all common styles like rules you've applied to `html`, `body`, `*`, etc to `common.scss` file

File: src/styles/card.scss [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+.card__img {
+  margin: 0 3px 40px;
+}
+
+.card__img-offer {
+  height: 134px;
+  width: 160px;
+}
```
Comment (Viktor-Kost) [inline src/styles/card.scss:28]: merge styles
```suggestion
.card__img-offer {
  display: block;
  margin: 0 auto 40px;
  height: 134px;
  width: 160px;
}
```

File: src/styles/card.scss [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+  font-family: Roboto;
+  src: url('https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,100;0,300;0,400;0,500;0,700;0,900;1,100;1,300;1,400;1,500;1,700;1,900&display=swap');
+}
+
+.body {
+  margin: 20px;
+}
+
+.card {
+  border: 1px solid #f3f3f3;
```
Comment (Viktor-Kost) [inline src/styles/card.scss:11]: you can add so your inner elements will be able to center with `margin: 0 auto`
```
display: flex;
flex-direction: column;
```

File: src/styles/card.scss [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+  font-weight: 700;
+  line-height: 16px;
+  text-align: center;
+  color: #fff;
+}
+
+.card__buy-button:hover {
+  border: 1px solid #00acdc;
+  background: #fff;
+  color: #00acdc;
```
Comment (Viktor-Kost) [inline src/styles/card.scss:94]: move repeated colors to varaibles

File: src/styles/card.scss [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+.card__price-value {
+  font-size: 16px;
+  font-weight: 700;
+  line-height: 18px;
+  color: #000;
+}
+
+.card__buy-button {
+  border: none;
+  width: 166px;
```
Comment (Viktor-Kost) [inline src/styles/card.scss:80]: full width
```suggestion
  width: 100%;
```

File: src/styles/card.scss [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+  display: flex;
+  font-weight: 500;
+  font-size: 12px;
+  line-height: 18px;
+  margin-bottom: 4px;
+}
+
+.card__code {
+  font-size: 10px;
+  font-weight: 400;
```
Comment (Viktor-Kost) [inline src/styles/card.scss:40]: it's a default browser value. no needs to redeclare it
```suggestion
```

File: src/styles/card.scss [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+
+.card__rate {
+  display: flex;
+  justify-content: space-between;
+  margin-bottom: 24px;
+}
+
+.card__reviews {
+  font-size: 10px;
+  font-weight: 400;
```
Comment (Viktor-Kost) [inline src/styles/card.scss:54]:
```suggestion
```

File: src/index.html [hunk 13]
```diff
@@ -12,7 +12,42 @@
       href="./styles/index.scss"
     />
   </head>
-  <body>
-    <h1>Product cards</h1>
+  <body class="body">
```
Comment (Viktor-Kost) [inline src/index.html:15]: no needs to add a class selector to a body
```suggestion
  <body>
```

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]

**Review by l4st1m0za — CHANGES_REQUESTED** [review]
Good job, but you need to try to pass all the test. Now, you have problems with font-weight, as I can see. As soon as you will reduce these problems, you can call the review once more!

**Review by l4st1m0za — APPROVED** [review]
Good job!

**Pull request closed without merging**